### PR TITLE
refactor(workers): Make reuse logic more consistent

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Use single `result` object for all return statements inside the function
- Introduce `isConflicted` flag for the result value

## Context:

- Ref #3866

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/branch/reuse.ts [hunk 1]
```diff
@@ -58,10 +60,12 @@ export async function shouldReuseExistingBranch(
       if (await isBranchModified(branchName)) {
         // TODO: Warn here so that it appears in PR body (#9720)
```
Comment (zharinov) [inline lib/workers/branch/reuse.ts:61]: #9720 🤔

File: lib/workers/branch/reuse.ts [hunk 2]
```diff
@@ -56,12 +58,13 @@ export async function shouldReuseExistingBranch(
       logger.debug(`Branch is stale and needs rebasing`);
       // We can rebase the branch only if no PR or PR can be rebased
       if (await isBranchModified(branchName)) {
-        // TODO: Warn here so that it appears in PR body (#9720)
```
Comment (zharinov) [inline lib/workers/branch/reuse.ts:59]: #9720

File: lib/workers/branch/reuse.ts [hunk 3]
```diff
@@ -71,17 +74,19 @@ export async function shouldReuseExistingBranch(
   }
 
   // Now check if PR is unmergeable. If so then we also rebase
-  if (pr?.isConflicted) {
+  result.isConflicted = pr?.isConflicted;
```
Comment (zharinov) [inline lib/workers/branch/reuse.ts:77]: This line is why we need these changes at all

Comment (rarkins) [inline lib/workers/branch/reuse.ts:77]:
```suggestion
  result.isConflicted = !!pr?.isConflicted;
```

So we can type it as Boolean later

**Review by rarkins — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**